controllers/snapshot/snapshot_adapter.go

```diff
@@ -130,6 +137,64 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 	return results.ContinueProcessing()
 }
 
+//EnsureCreationOfEnvironment makes sure that all envrionemnts that were requested via
+//IntegrationTestScenarios get created, in case that environment is already created, provides
+//a message about this fact
+func (a *Adapter) EnsureCreationOfEnvironment() (results.OperationResult, error) {
+
+	integrationTestScenarios, err := helpers.GetAllIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+
+	if err != nil {
+		a.logger.Error(err, "Failed to get Integration test scenarios for following application",
+			"Application.Name", a.application.Name,
+			"Application.Namespace", a.application.Namespace)
+		return results.RequeueOnErrorOrStop(err)
+	}
+
+	allEnvironments, err := a.getAllEnvironments()
+	if err != nil {
+		return results.RequeueOnErrorOrStop(err)
+	}
+
+	if integrationTestScenarios != nil {
+		for _, integrationTestScenario := range *integrationTestScenarios {
+			integrationTestScenario := integrationTestScenario //G601
+			for _, environment := range *allEnvironments {
+				environment := environment //G601
+				//prevent creating already existing environments
+				if helpers.HasLabelWithValue(&environment, "test.appstudio.openshift.io/snapshot", a.snapshot.Name) && helpers.HasLabelWithValue(&environment, "test.appstudio.openshift.io/scenario", integrationTestScenario.Name) {
+					fmt.Println("ENV: " + environment.Name + "Already exists and contains snapshot: " + a.snapshot.Name + " and scenario: " + integrationTestScenario.Name)
+					return results.ContinueProcessing()
+				}
+			}
+			if integrationTestScenario.Spec.Environment.Name != "" {
```

Comment:
It might be a good idea to have an inverse check of this at the beginning of the [for loop](https://github.com/redhat-appstudio/integration-service/pull/74/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R160) and have it `continue` if the integrationTestScenario doesn't have an environment associated with it.
This way we wouldn't be checking if an environment exists for scenarios that don't require an environment.

Reply:
Also the same comment as above for just checking `integrationTestScenario.Spec.Environment` instead of the `Name` applies.

Reply:
Does the updated solution suits the description?

Reply:
Yes, the updated solution looks good